### DISCUSSION TITLE
Vim extension slow

### DISCUSSION THREAD

**Discussion by PhotonQuantum** [coder/code-server#2828] [Q&A] [answered]
Yes, I've seen the related issue #308. I know that code-server is now running vsvim client-side to mitigate this problem.
However, from my experience, my editor starts lagging once I activate the vim plugin, so I'm suspecting this feature isn't working correctly on my installation.

I checked my Worker Extension Host log and got
```
[2021-03-07 02:25:43.472] [exthost.log] [info] extension host started
[2021-03-07 02:25:43.491] [exthost.log] [info] eager extensions activated
```

And this is what I got from Remote Extension Host:
```
[2021-03-06 18:36:50.319] [exthost] [info] ExtensionService#_doActivateExtension vscodevim.vim {"startup":true,"extensionId":{"value":"vscodevim.vim","_lower":"vscodevim.vim"},"activationEvent":"*"}
[2021-03-06 18:36:50.319] [exthost] [info] ExtensionService#loadCommonJSModule file:///home/coder/vscode_data/extensions/vscodevim.vim-1.18.9/out/extension
```

I noticed that my Chrome console has the following warnings and errors:
```
[Service Worker] registered
Unchecked runtime.lastError: The message port closed before a response was received.
The web worker extension host is started without an iframe sandbox!
```

You may reproduce my installation by this docker image: https://github.com/PhotonQuantum/latex-workshop-docker.

I'm not sure what's going on, so these logs may not be helpful. I'm willing to provide more diagnostic info if needed.

**Comment by code-asher** [comment, marked as answer]
Ohh interesting, I think the problem is that we used to have our own web/client extension implementation that loaded the extension from disk into the browser.

We removed that because VS Code added their own implementation. But theirs doesn't load from disk, instead it pulls directly from the marketplace when installed from the UI.

So installing with `--install-extension` ends up leaving it running on the server and no longer loads  it into the browser.

I think currently the only workaround then is to delete the Vim extension from disk then install it from VS Code's UI.

**Reply by code-asher** [reply]
We'll have to look into seeing whether we can make this work as it did before.

**Reply by PhotonQuantum** [reply]
I installed the plugin again. Now vsvim works in the browser and the input is smooth. Thanks!
